homeassistant/helpers/config_validation.py

```diff
@@ -745,11 +739,20 @@ def deprecated(
             " please remove it from your configuration"
         )
 
+    if fail:
```

Comment:
I don't think we need to both print an error and raise a validation error, let's just raise.

Reply:
done, added an extra test